### ISSUE TITLE
[battery_plus] isInBatterySaveMode fails will NullPointerException

### ISSUE THREAD

**Issue by bejaeger** [fluttercommunity/plus_plugins#713]
## System info

Using a Samsung S21 (SM-G991W)
Platform the Issue occurs on: android 12 (targetSdkVersion 31)
Plugin name: battery_plus
Plugin version: 2.1.0

## Steps to Reproduce

```dart
      final Battery battery = Battery();
      final isInBatterySaveMode = await battery.isInBatterySaveMode;
```
yields to the following error message
```
PlatformException(error, Attempt to invoke virtual method 'boolean java.lang.String.equals(java.lang.Object)' on a null object reference, null, java.lang.NullPointerException: Attempt to invoke virtual method 'boolean java.lang.String.equals(java.lang.Object)' on a null object reference
I/flutter (22574): 	at dev.fluttercommunity.plus.battery.BatteryPlusPlugin.getPowerSaveModeSamsung(BatteryPlusPlugin.java:167)
...
```

<details>
<summary>Logs</summary>

```dart
[  +15 ms] E/MethodChannel#dev.fluttercommunity.plus/battery(29604): Failed to handle method call
[        ] E/MethodChannel#dev.fluttercommunity.plus/battery(29604): java.lang.NullPointerException: Attempt to invoke virtual method 'boolean java.lang.String.equals(java.lang.Object)' on a null object reference
[        ] E/MethodChannel#dev.fluttercommunity.plus/battery(29604):    at dev.fluttercommunity.plus.battery.BatteryPlusPlugin.getPowerSaveModeSamsung(BatteryPlusPlugin.java:167)
[        ] E/MethodChannel#dev.fluttercommunity.plus/battery(29604):    at dev.fluttercommunity.plus.battery.BatteryPlusPlugin.isInPowerSaveMode(BatteryPlusPlugin.java:154)
[        ] E/MethodChannel#dev.fluttercommunity.plus/battery(29604):    at dev.fluttercommunity.plus.battery.BatteryPlusPlugin.onMethodCall(BatteryPlusPlugin.java:85)
[        ] E/MethodChannel#dev.fluttercommunity.plus/battery(29604):    at io.flutter.plugin.common.MethodChannel$IncomingMethodCallHandler.onMessage(MethodChannel.java:262)
[        ] E/MethodChannel#dev.fluttercommunity.plus/battery(29604):    at io.flutter.embedding.engine.dart.DartMessenger.invokeHandler(DartMessenger.java:178)
[        ] E/MethodChannel#dev.fluttercommunity.plus/battery(29604):    at io.flutter.embedding.engine.dart.DartMessenger.lambda$handleMessageFromDart$0$DartMessenger(DartMessenger.java:206)
[        ] E/MethodChannel#dev.fluttercommunity.plus/battery(29604):    at io.flutter.embedding.engine.dart.-$$Lambda$DartMessenger$6ZD1MYkhaLxyPjtoFDxe45u43DI.run(Unknown Source:12)
[        ] E/MethodChannel#dev.fluttercommunity.plus/battery(29604):    at android.os.Handler.handleCallback(Handler.java:938)
[        ] E/MethodChannel#dev.fluttercommunity.plus/battery(29604):    at android.os.Handler.dispatchMessage(Handler.java:99)
[        ] E/MethodChannel#dev.fluttercommunity.plus/battery(29604):    at android.os.Looper.loopOnce(Looper.java:226)
[        ] E/MethodChannel#dev.fluttercommunity.plus/battery(29604):    at android.os.Looper.loop(Looper.java:313)
[        ] E/MethodChannel#dev.fluttercommunity.plus/battery(29604):    at android.app.ActivityThread.main(ActivityThread.java:8582)
[        ] E/MethodChannel#dev.fluttercommunity.plus/battery(29604):    at java.lang.reflect.Method.invoke(Native Method)
[        ] E/MethodChannel#dev.fluttercommunity.plus/battery(29604):    at com.android.internal.os.RuntimeInit$MethodAndArgsCaller.run(RuntimeInit.java:563)
[        ] E/MethodChannel#dev.fluttercommunity.plus/battery(29604):    at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:1133)
[   +2 ms] E/flutter (29604): [ERROR:flutter/lib/ui/ui_dart_state.cc(209)] Unhandled Exception: PlatformException(error, Attempt to invoke virtual method 'boolean java.lang.String.equals(java.lang.Object)' on a null object reference, null, java.lang.NullPointerException: Attempt to invoke virtual method 'boolean java.lang.String.equals(java.lang.Object)' on a null object reference
[        ] E/flutter (29604):   at dev.fluttercommunity.plus.battery.BatteryPlusPlugin.getPowerSaveModeSamsung(BatteryPlusPlugin.java:167)
[        ] E/flutter (29604):   at dev.fluttercommunity.plus.battery.BatteryPlusPlugin.isInPowerSaveMode(BatteryPlusPlugin.java:154)
[        ] E/flutter (29604):   at dev.fluttercommunity.plus.battery.BatteryPlusPlugin.onMethodCall(BatteryPlusPlugin.java:85)
[        ] E/flutter (29604):   at io.flutter.plugin.common.MethodChannel$IncomingMethodCallHandler.onMessage(MethodChannel.java:262)
[        ] E/flutter (29604):   at io.flutter.embedding.engine.dart.DartMessenger.invokeHandler(DartMessenger.java:178)
[        ] E/flutter (29604):   at io.flutter.embedding.engine.dart.DartMessenger.lambda$handleMessageFromDart$0$DartMessenger(DartMessenger.java:206)
[        ] E/flutter (29604):   at io.flutter.embedding.engine.dart.-$$Lambda$DartMessenger$6ZD1MYkhaLxyPjtoFDxe45u43DI.run(Unknown Source:12)
[        ] E/flutter (29604):   at android.os.Handler.handleCallback(Handler.java:938)
[        ] E/flutter (29604):   at android.os.Handler.dispatchMessage(Handler.java:99)
[        ] E/flutter (29604):   at android.os.Looper.loopOnce(Looper.java:226)
[        ] E/flutter (29604):   at android.os.Looper.loop(Looper.java:313)
[        ] E/flutter (29604):   at android.app.ActivityThread.main(ActivityThread.java:8582)
[        ] E/flutter (29604):   at java.lang.reflect.Method.invoke(Native Method)
[        ] E/flutter (29604):   at com.android.internal.os.RuntimeInit$MethodAndArgsCaller.run(RuntimeInit.java:563)
[        ] E/flutter (29604):   at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:1133)
[        ] E/flutter (29604): )
[        ] E/flutter (29604): #0      StandardMethodCodec.decodeEnvelope (package:flutter/src/services/message_codecs.dart:607:7)
[        ] E/flutter (29604): #1      MethodChannel._invokeMethod (package:flutter/src/services/platform_channel.dart:167:18)
[        ] E/flutter (29604): <asynchronous suspension>
[        ] E/flutter (29604): #2      QuestViewModel.checkIfBatterySaveModeOn (package:afkcredits/ui/views/common_viewmodels/quest_viewmodel.dart:158:33)
[        ] E/flutter (29604): <asynchronous suspension>
[        ] E/flutter (29604): #3      QuestViewModel.startQuestMain (package:afkcredits/ui/views/common_viewmodels/quest_viewmodel.dart:122:5)
[        ] E/flutter (29604): <asynchronous suspension>
[        ] E/flutter (29604): #4      ActiveMapQuestViewModel.maybeStartQuest (package:afkcredits/ui/views/active_map_quest/active_map_quest_viewmodel.dart:58:11)
[        ] E/flutter (29604): <asynchronous suspension>
[        ] E/flutter (29604): 
[ +155 ms] I/Counters(29604): exceeded sample count in FrameTime
[+5938 ms] I/Counters(29604): exceeded sample count in FrameTime
[+5094 ms] I/Counters(29604): exceeded sample count in FrameTime
[+2227 ms] I/Counters(29604): exceeded sample count in FrameTime
[+1142 ms] I/Counters(29604): exceeded sample count in FrameTime
[+4601 ms] Application finished.

```

</details>

<details>
<summary>Flutter doctor output</summary>

```
[✓] Flutter (Channel stable, 2.8.1, on macOS 11.5.2 20G95 darwin-x64, locale en-CA)
    • Flutter version 2.8.1 at /usr/local/Cellar/flutter
    • Upstream repository https://github.com/flutter/flutter.git
    • Framework revision 77d935af4d (6 weeks ago), 2021-12-16 08:37:33 -0800
    • Engine revision 890a5fca2e
    • Dart version 2.15.1

[✓] Android toolchain - develop for Android devices (Android SDK version 30.0.2)
    • Android SDK at /Users/bj/Library/Android/sdk
    • Platform android-31, build-tools 30.0.2
    • Java binary at: /Applications/Android Studio.app/Contents/jre/jdk/Contents/Home/bin/java
    • Java version OpenJDK Runtime Environment (build 1.8.0_242-release-1644-b3-6222593)
    • All Android licenses accepted.

[!] Xcode - develop for iOS and macOS (Xcode 12.5.1)
    • Xcode at /Applications/Xcode.app/Contents/Developer
    ! Flutter recommends a minimum Xcode version of 13.0.0.
      Download the latest version or update via the Mac App Store.
    • CocoaPods version 1.11.2

[✓] Chrome - develop for the web
    • Chrome at /Applications/Google Chrome.app/Contents/MacOS/Google Chrome

[✓] Android Studio (version 4.0)
    • Android Studio at /Applications/Android Studio.app/Contents
    • Flutter plugin version 46.0.2
    • Dart plugin version 193.7361
    • Java version OpenJDK Runtime Environment (build 1.8.0_242-release-1644-b3-6222593)

[✓] VS Code (version 1.63.2)
    • VS Code at /Applications/Visual Studio Code.app/Contents
    • Flutter extension version 3.32.0

[✓] Connected device (2 available)
    • SM G991W (mobile) • R5CR605LX1Y • android-arm64  • Android 12 (API 31)
    • Chrome (web)      • chrome      • web-javascript • Google Chrome 97.0.4692.99

! Doctor found issues in 1 category.
```

</details>

**Comment by vbuberen** [comment]
Thanks for your report. I was able to reproduce the issue on another Samsung with Android 10. Will look at the reason in nearest time.

**Comment by vbuberen** [comment]
Fix released in `2.1.1`. Please give it a try and share feedback if something is still wrong.
